examples/upgradeable-contract/upgradeable-flipper/upgradeable.rs

```diff
@@ -0,0 +1,178 @@
+use core::marker::PhantomData;
+use ink_primitives::{
+    Key,
+    KeyPtr,
+};
+use ink_storage::traits::{
+    PackedAllocate,
+    PackedLayout,
+    SpreadAllocate,
+    SpreadLayout,
+};
+use scale::{
+    Decode,
+    Encode,
+};
+
+#[derive(Debug)]
+pub struct Initialized;
+#[derive(Debug)]
+pub struct NotInitialized;
+
+#[derive(Debug, Decode, Encode)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub struct Upgradeable<T: PackedLayout, InitializationStatus = Initialized> {
```

Comment:
Could you add some doc comments to these `pub`'s and a comment on why the `PackedLayout` / `SpreadLayout` implementations all have to be implemented? I think something like the comment by Alex would already be sufficient:

>Yeah all of this boilerplate is only needed because the current system doesn't allow to change the contract root key without manually implementing SpreadLayout.